duckduckgo/Android
# Allow encoded spaces in url path

Task/Issue URL: https://github.com/duckduckgo/Android/issues/476
Tech Design URL: N/A

**Description**:
1. fix urls like http://www.example.com/pa%20th leading to search query. spaces shouldn't be in encoded uriString, but decoded path can have spaces
1. Add unit test to prove valid urls like http://www.example.com/pa%20th are treated like web urls


**Steps to test this PR**:
1. Type `http://www.example.com/pa%20th` in top address bar and submit search
1. see the webpage with title `Example Domain` displayed
1. Alternatively run new unit test

![dgg-issue476](https://user-images.githubusercontent.com/8261416/59153335-ae6d6d80-8a4e-11e9-8dcd-f68d663b9bf2.gif)

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
